app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt

```diff
@@ -1265,10 +1266,19 @@ class LibraryController(
     }
 
     fun search(query: String?): Boolean {
+        if (!query.isNullOrBlank() && this.query.isBlank() && !presenter.showAllCategories) {
+            oldShowAllCategories = presenter.showAllCategories
+            preferences.showAllCategories().set(true)
+            presenter.getLibrary()
+        } else if (query.isNullOrBlank() && this.query.isNotBlank() && !oldShowAllCategories) {
+            preferences.showAllCategories().set(oldShowAllCategories)
+            presenter.getLibrary()
+        }
+
         if (query != this.query && !query.isNullOrBlank()) {
             binding.libraryGridRecycler.recycler.scrollToPosition(0)
         }
-        this.query = query ?: ""
+        this.query = (query ?: "").replace("'", "’")
```

Comment:
Don't do that, `U+0027` apostrophe is much more used that's why `U+2019` is changed to this one for Global Search. But yeah maybe it's not necessary for Library Search.
I think it's best to do one of the following options:
1. Replace apostrophe type only for Global Search: Remove the `toNormalized()` in the `setOnClickListener` and `setOnLongClickListener` on `title` (`init` of `MangaHeaderHolder`). Then add `toNormalized()` in the `globalSearch()` method in `MangaDetailsController`
2. Or normalize the query and titles in library during the filter: In the `filter()` method in `LibraryItem` add a `toNormalized()` to the `manga.title` and the query where the contains check is done
Pros of this one are that it would work if you have both apostrophes type in your library. Cons are that you should probably do the same thing for the `Recents` tab to be consistent.

Not sure which one is the best option :shrug:

Reply:
Since my approach is only affecting library search and has no performance downside regarding replacement I don't see why is there a need for another approach?

Reply:
Your approach doesn't solve the problem it just reverses it. What if you have a manga with `U+0027` (the one you replace) apostrophe in your library ? You won't be able to search it, exactly like before for the other apostrophe type

Reply:
Oh I see now. I will take a look at what you suggested earlier.